import "door_status.proto";
import "position_status.proto";
import "vehicle_status.proto";
import "window_status.proto";
import "tyre_status.proto";
import "soc_status.proto";
import "alarm_signal.proto";
import "driving_behaviour_msg.proto";

option java_package = "com.nextev.cvs_proto.protobuf.events";
option java_outer_classname = "ECallMessage";

message ECallEvent {
    optional string id = 1;            // unique id, VIN for vehicle data
    optional int32 version = 2;         // event protocol version
    optional uint64 sample_ts = 3;      // event unix timestamp, when call service is triggered
    optional uint64 event_id = 4;       // event id, currenttimestamp - millisecond
    optional string reason_code = 5;    // the reason of call service is trigger, seperated by underscore
    optional ECallStatus status = 6;    // some important vehicle status
}

message ECallStatus {
    optional PositionStatus position_status = 1;
    optional VehicleStatus vehicle_status = 2;
    optional DoorStatus door_status = 3;
    optional WindowStatus window_status = 4;
    optional TyreStatus tyre_status = 5;
    optional SOCStatus soc_status = 6;
    optional AlarmSignal alarm_signal = 7;
    optional DrivingBehaviourEvent driving_behaviour_event = 8;
}
